src/dmd/common/file.d

```diff
@@ -52,6 +54,8 @@ struct FileMapping(Datum)
     private const(char)* name;
     // state }
 
+  nothrow:
```

Comment:
I agree with all of them except possibly this one. 

The constructor call `exit(1)` on failure to map. 
This should probably be made to be a call to `fatal()`, which is in the process of being made more "DMD as a library" friendly see https://github.com/dlang/dmd/pull/13891

Reply:
It's a good point, but it doesn't belong in this PR, which is only about `nothrow`.